src/postgres/src/backend/executor/ybcModifyTable.c

```diff
@@ -265,14 +265,20 @@ static void YBCExecWriteStmt(YBCPgStatement ybc_stmt,
 }
 
 static Oid YBCApplyInsertRow(
-    YBCPgStatement insert_stmt, Relation rel, TupleDesc tupleDesc,
-    HeapTuple tuple, OnConflictAction onConflictAction, Datum *ybctid,
+    YBCPgStatement insert_stmt, Relation rel, TupleTableSlot *slot, 
+    OnConflictAction onConflictAction, Datum *ybctid,
     YBCPgTransactionSetting transaction_setting) {
 	Oid            relfileNodeId    = YbGetRelfileNodeId(rel);
 	AttrNumber     minattr          = YBGetFirstLowInvalidAttributeNumber(rel);
 	int            natts            = RelationGetNumberOfAttributes(rel);
 	Bitmapset      *pkey            = YBGetTablePrimaryKeyBms(rel);
+	TupleDesc		tupleDesc       = slot->tts_tupleDescriptor;
+	HeapTuple 	    tuple           = slot->tts_tuple;
```

Comment:
Based on the definition of ExecMaterializeSlot, I haven't used it. So, ybctid and oid are updated in tuple itself.